src/Database/Type/DateTimeType.php

```diff
@@ -179,18 +194,20 @@ public function toPHP($value, DriverInterface $driver)
             return null;
         }
 
-        $instance = clone $this->_datetimeInstance;
+        $class = $this->_className;
         if (is_int($value)) {
-            $instance = $instance->setTimestamp($value);
+            $instance = new $class('@' . $value);
         } else {
             if (strpos($value, '0000-00-00') === 0) {
                 return null;
             }
-            $instance = $instance->modify($value);
+            $instance = new $class($value, $this->dbTimezone);
         }
 
-        if ($instance->getTimezone()->getName() !== date_default_timezone_get()) {
-            $instance = $instance->setTimezone(new DateTimeZone(date_default_timezone_get()));
+        if (!$this->keepDatabaseTimezone) {
+            if ($instance->getTimezone()->getName() !== $this->defaultTimezone->getName()) {
```

Comment:
This nesting can be avoided by hoisting the check in outer `if` block.

Reply:
I will change this, but I'm wondering if nested checks can look cleaner than multi-line if checks sometimes.

```
        if (
            !$this->keepDatabaseTimezone &&
            $instance->getTimezone()->getName() !== $this->defaultTimezone->getName()
        ) {
```

Reply:
Asthetics are not really a concern here as the nested `if` creates an unnecessary code branch.

Reply:
That's pretty pedantic lol.